### PR TITLE
feat: validate git version

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Updates development docs and adds a test

## Context:

Helpful for #9938

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/git/index.spec.ts [hunk 1]
```diff
@@ -87,6 +88,12 @@ describe(getName(), () => {
     await base.cleanup();
   });
 
+  it('has minimum git version', async () => {
+    const res = await git.gitVersion();
+    // git show-current was introduced in 2.22.0
+    expect(semver.isVersion).toBeTrue();
+    expect(semver.isGreaterThan(res, '2.21.0')).toBeTrue();
```
Comment (viceice) [inline lib/util/git/index.spec.ts:95]:
```suggestion
    expect(semver.isGreaterThan(res, '2.22.0')).toBeTrue();
```

?

**Review by viceice — COMMENTED** [review]
Maybe also log as warning at runtime? So if a user is using a custom docker image or the plain npm package?

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -140,6 +141,23 @@ let gitInitialized: boolean;
 
 let privateKeySet = false;
 
+export async function gitVersion(): Promise<string> {
+  const globalGit = Git();
+  let res = null;
+  try {
+    const raw = await globalGit.raw(['--version']);
```
Comment (viceice) [inline lib/util/git/index.ts:148]: Didn't simple-git have a helper for it?

File: lib/util/git/index.spec.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/git/index.spec.ts:95]: 2.22.0 is ok. Maybe need to add an equals ||

File: lib/util/git/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/util/git/index.ts:148]: I couldn't find it

File: lib/util/git/index.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/util/git/index.spec.ts:95]: yes, better use greater or equal

File: lib/util/git/index.ts [hunk 3]
```diff
@@ -141,6 +142,42 @@ let gitInitialized: boolean;
...
 let privateKeySet = false;
 
+export const GIT_MINIMUM_VERSION = '2.22.0'; // git show-current
+
+export async function validateGitVersion(): Promise<boolean> {
+  let version: string;
+  const globalGit = Git();
+  try {
+    const raw = await globalGit.raw(['--version']);
+    console.warn(raw); // eslint-disable-line
```
Comment (viceice) [inline lib/util/git/index.ts:152]: ? Leftover from testing?

**Review by viceice — APPROVED** [review]

**Pull request merged**